### PR TITLE
Hw 1 solution

### PR DESCRIPTION
**Pull request by ClubJamm**

### CODE REVIEW

File: src/main/java/core/basesyntax/Main.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+        for (int i = 0; i < figures.length / 2; i++) {
+            figures[i] = figureSupplier.getRandomFigure();
+        }
+
+        for (int i = figures.length / 2; i < figures.length; i++) {
+            figures[i] = figureSupplier.getRandomFigure();
+        }
+
+        for (Figure figure : figures) {
+            figure.draw();
```
Comment (Vasul-Pankiv) [inline src/main/java/core/basesyntax/Main.java:17]: call draw method while you creating figures

**Review by Vasul-Pankiv — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/Circle.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    private double radius;
+
+    public Circle(String color, double radius) {
+        super(color);
+        this.radius = radius;
+    }
+
+    @Override
+    public double getArea() {
+        return Math.PI * (radius * radius);
```
Comment (sarakhmen) [inline src/main/java/core/basesyntax/Circle.java:13]:
```suggestion
        return Math.PI * radius * radius;
```

File: src/main/java/core/basesyntax/ColorSupplier.java [hunk 3]
```diff
@@ -0,0 +1,12 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class ColorSupplier {
+    private String[] colors = {"red", "blue", "green", "yellow", "pink", "white", "black"};
```
Comment (sarakhmen) [inline src/main/java/core/basesyntax/ColorSupplier.java:6]: let's create an enum for colors

File: src/main/java/core/basesyntax/FIgureSupplier.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FIgureSupplier {
+    private ColorSupplier colorSupplier = new ColorSupplier();
+    private Random random = new Random();
+
+    public Figure getRandomFigure() {
+        int figureType = random.nextInt(5);
```
Comment (sarakhmen) [inline src/main/java/core/basesyntax/FIgureSupplier.java:10]:
```suggestion
        int figureType = random.nextInt(5);
```
let's create a constant for this magic number

File: src/main/java/core/basesyntax/FIgureSupplier.java [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+        String color = colorSupplier.getRandomColor();
+
+        switch (figureType) {
+            case 0:
+                return new Square(color,
+                        random.nextDouble() * 10 + 1);
```
Comment (sarakhmen) [inline src/main/java/core/basesyntax/FIgureSupplier.java:16]: why do you apply "*10+1"? And why is it duplicated so much times?

File: src/main/java/core/basesyntax/FIgureSupplier.java [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+                        random.nextDouble() * 10 + 1,
+                        random.nextDouble() * 10 + 1,
+                        random.nextDouble() * 10 + 1);
+            default:
+                return getRandomFigure();
+        }
+    }
+
+    public Figure getDefaultFigure() {
+        return new Circle("white", 10);
```
Comment (sarakhmen) [inline src/main/java/core/basesyntax/FIgureSupplier.java:39]: magic number and magic word :)

**Review by sarakhmen — CHANGES_REQUESTED** [review]
let's improve your solution :)

File: src/main/java/core/basesyntax/ColorSupplier.java [hunk 7]
```diff
@@ -0,0 +1,12 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class ColorSupplier {
+    private Random random = new Random();
+
+    public Color getRandomColor() {
```
Comment (Elena-Bruyako) [inline src/main/java/core/basesyntax/ColorSupplier.java:8]:
```suggestion
    public String getRandomColor() {
```

File: src/main/java/core/basesyntax/FIgureSupplier.java [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+import java.util.Random;
+
+public class FIgureSupplier {
+    private static final int NUMBER_OF_FIGURE_TYPES = 5;
+    private ColorSupplier colorSupplier = new ColorSupplier();
+    private Random random = new Random();
+
+    public Figure getRandomFigure() {
+        int figureType = random.nextInt(NUMBER_OF_FIGURE_TYPES);
+        Color color = colorSupplier.getRandomColor();
```
Comment (Elena-Bruyako) [inline src/main/java/core/basesyntax/FIgureSupplier.java:12]:
```suggestion
        String color = colorSupplier.getRandomColor();
```

File: src/main/java/core/basesyntax/Figure.java [hunk 9]
```diff
@@ -0,0 +1,13 @@
+package core.basesyntax;
+
+public abstract class Figure implements Drawable, AreaCalculable {
+    protected Color color;
+
+    public Figure(Color color) {
+        this.color = color;
+    }
+
+    public abstract double getArea();
```
Comment (Elena-Bruyako) [inline src/main/java/core/basesyntax/Figure.java:10]: remove, you implemented interface with this method and every type of figure that extends Figure should override this method

File: src/main/java/core/basesyntax/Figure.java [hunk 10]
```diff
@@ -0,0 +1,13 @@
...
+
+    public abstract void draw();
```
Comment (Elena-Bruyako) [inline src/main/java/core/basesyntax/Figure.java:12]: remove, you implemented interface with this method and every type of figure that extends Figure should override this method

File: src/main/java/core/basesyntax/Figure.java [hunk 11]
```diff
@@ -0,0 +1,13 @@
+package core.basesyntax;
+
+public abstract class Figure implements Drawable, AreaCalculable {
+    protected Color color;
```
Comment (Elena-Bruyako) [inline src/main/java/core/basesyntax/Figure.java:4]:
```suggestion
    protected String color;
```

File: src/main/java/core/basesyntax/FIgureSupplier.java [hunk 12]
```diff
@@ -0,0 +1,36 @@
...
+                    getRandomSize());
+            case 3 -> new RightTriangle(color,
+                    getRandomSize(), getRandomSize());
+            case 4 -> new IsoscelesTrapezoid(color,
+                    getRandomSize(), getRandomSize(), getRandomSize());
+            default -> getDefaultFigure();
+        };
+    }
+
+    private double getRandomSize() {
```
Comment (Elena-Bruyako) [inline src/main/java/core/basesyntax/FIgureSupplier.java:29]: private method should be placed after public method

File: src/main/java/core/basesyntax/FIgureSupplier.java [hunk 13]
```diff
@@ -0,0 +1,36 @@
...
+        return random.nextDouble();
+    }
+
+    public Figure getDefaultFigure() {
+        return new Circle(Color.WHITE, 10);
```
Comment (Elena-Bruyako) [inline src/main/java/core/basesyntax/FIgureSupplier.java:34]: Make Color.WHITE, 10 as constants

File: src/main/java/core/basesyntax/Main.java [hunk 14]
```diff
@@ -0,0 +1,18 @@
+package core.basesyntax;
+
+public class Main {
+    public static void main(String[] args) {
+        FIgureSupplier figureSupplier = new FIgureSupplier();
+        Figure[] figures = new Figure[6];
```
Comment (Elena-Bruyako) [inline src/main/java/core/basesyntax/Main.java:6]: 6 is a magic number, create constant

File: src/main/java/core/basesyntax/Main.java [hunk 15]
```diff
@@ -0,0 +1,18 @@
...
+
+        for (int i = 0; i < figures.length / 2; i++) {
```
Comment (Elena-Bruyako) [inline src/main/java/core/basesyntax/Main.java:8]: avoid 2 for loops. think how to implement task with one for loop

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]

**Comment by ClubJamm** [comment]
One of mentors says create enum Colors and another mentor says use String instead of Colors. I don't understand)

File: src/main/java/core/basesyntax/ColorSupplier.java (hunk 7 above)
Comment (Elena-Bruyako) [reply src/main/java/core/basesyntax/ColorSupplier.java:8]: Let's see task description

In the main() method we need to create an array of figures (the size of the array can be 3 or 6, it doesn't matter). The first half of figures in this array should be generated with random parameters.

For this purpose create two more classes:

`ColorSupplier with public String getRandomColor() method - for generating random color,`

Creating enum helps you to provide different colors, but method String getRandomColor() should return String representation of Color (from enum)

File: src/main/java/core/basesyntax/FIgureSupplier.java (hunk 12 above)
Comment (Elena-Bruyako) [reply src/main/java/core/basesyntax/FIgureSupplier.java:32]: not fixed

File: src/main/java/core/basesyntax/FIgureSupplier.java [hunk 16]
```diff
@@ -0,0 +1,52 @@
...
+
+    private double getRandomSize() {
+        return random.nextDouble();
+    }
+
+    public Figure getDefaultFigure() {
+        return new Circle(DEFAULT_COLOR, DEFAULT_RADIUS);
+    }
+
+    public Figure[] generateFigures() {
```
Comment (Elena-Bruyako) [inline src/main/java/core/basesyntax/FIgureSupplier.java:40]: This logic is not related to FigureSupplier class, so replace it to Main class

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
Answered your question. Please, fix comments
